test/functional/mempool_fee_histogram.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test mempool fee histogram."""
+
+from decimal import Decimal
+
+from test_framework.blocktools import COINBASE_MATURITY
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_greater_than,
+    assert_greater_than_or_equal,
+    assert_no_key,
+)
+
+class MempoolFeeHistogramTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def run_test(self):
+        node = self.nodes[0]
+        node.generate(COINBASE_MATURITY + 2)
```

Comment:
Test is failing here. `TypeError: generatetoaddress() missing 1 required keyword-only argument: 'invalid_call'`. I think there is something wrong with the implementation of this function in `test_node` and I don't see any other test calling `generate` from node, I will take a look on it. But I think you can use `generatetoaddress` from `test_framework`.

Reply:
Yes, that's also a reason why I converted the PR to a draft. I have done as much as I could today. I will continue tomorrow.

Reply:
Rebased because of #23300.

Reply:
Great. #23300 solves it!